### ISSUE TITLE
Better Validation for Test Cases

### ISSUE THREAD

**Issue by cjol** [gsmainclusivetechlab/interop-test-platform#383]
When uploading YAML test cases, it's possible to make a mistake and upload invalid data. This should be validated before the database is updated, otherwise this happens:

![image](https://user-images.githubusercontent.com/1517675/104012720-33c77f80-51a8-11eb-852d-2beb51258b76.png)

Notice that this is the top-level test cases page, which is now inaccessible making it difficult to fix the issue. The problem was the following row in the database, where you can see that the `method` field is missing from the request object.

```
+------+--------------+-----------+-----------+-------------+---------+--------+-----------+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+---------------------+---------------------+
| id   | test_case_id | source_id | target_id | api_spec_id | path    | method | pattern   | trigger           | request                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | response                                                                                                                                                                                                                                                                                                                                                                                                                                                           | position | created_at          | updated_at          |
+------+--------------+-----------+-----------+-------------+---------+--------+-----------+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+---------------------+---------------------+
| 1047 |          146 |         3 |         2 |           2 | /quotes | POST   | ^quotes/$ | {"amount": "100"} | {"body": {"note": "hej", "payee": {"partyIdInfo": {"fspId": "testfsp4", "partyIdType": "MSISDN", "partyIdentifier": "{{ TESTFSP4_MSISDN }}"}}, "payer": {"partyIdInfo": {"fspId": "testfsp3", "partyIdType": "MSISDN", "partyIdentifier": "{{ TESTFSP3_MSISDN }}"}, "personalInfo": {"complexName": {"lastName": "Hagman", "firstName": "Mats"}, "dateOfBirth": "1983-10-25"}}, "amount": {"amount": "10", "currency": "EUR"}, "quoteId": "{{ QUOTE_ID }}", "amountType": "SEND", "extensionList": {"extension": [{"key": "KYCPayerTier", "value": "1"}, {"key": "KYCNationality", "value": "CI"}, {"key": "KYCOccupation", "value": "Engineer"}, {"key": "KYCEmployerName", "value": "Example Corp"}, {"key": "KYCContactPhone", "value": "1122334455"}, {"key": "KYCEmailAddress", "value": "user@mail.com"}, {"key": "KYCBirthCountry", "value": "CI"}, {"key": "KYCPayerAddress1", "value": "Immeuble Le Quartz Boulevard Val�ry Giscard d'Estaing Marcory"}, {"key": "KYCPayerAddress2", "value": "11 BP 202 Abidjan 11"}, {"key": "KYCPayerAddressCity", "value": "Abidjan"}, {"key": "KYCPayerAddressCode", "value": "NE1 3TQ"}, {"key": "KYCPayerIDType", "value": "PASSPORT"}, {"key": "KYCPayerIDValue", "value": "770423742"}, {"key": "KYCPayerTransferReason", "value": "Bill payment"}]}, "transactionId": "{{ TRANSACTION_ID }}", "transactionType": {"scenario": "TRANSFER", "initiator": "PAYER", "initiatorType": "CONSUMER"}}, "status": "202", "headers": {"Date": ["{{ \"now\" | date(\"F j, Y, g:i a\") }}"], "Content-Type": ["application/vnd.interoperability.quotes+json;version=1.0"]}} | {"uri": "/quotes", "body": {"Request Body": "{{quoteRequest}}"}, "method": "POST", "headers": {"Date": ["{{ \"now\" | date(\"F j, Y, g:i a\") }}"], "Accept": ["application/vnd.interoperability.quotes+json;version=1"], "FSPIOP-URI": ["/quotes"], "Content-Type": ["application/vnd.interoperability.quotes+json;version=1.0"], "FSPIOP-Source": ["testfsp3"], "FSPIOP-Signature": [null], "FSPIOP-Destination": ["testfsp4"], "FSPIOP-HTTP-Method": ["POST"]}} |        1 | 2021-01-08 11:24:58 | 2021-01-08 11:24:58 |
+------+--------------+-----------+-----------+-------------+---------+--------+-----------+-------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+---------------------+---------------------+
1 row in set (0.00 sec)

```

**Comment by SamyBouch** [comment]
Additionally it would be nice if there was some Twig template validation when importing a test case. For example in this test run https://compliance.mobilemoneyapi.io/sessions/8/test-cases/240/test-runs/266 the URL didn't generate a UUID because I had put `{ uuidv4() }` instead of `{{ uuidv4() }}` in the yaml file so it would be helpful if these sort of errors were flagged when importing.

I also had an issue when I imported a separate test case. The admin/test-cases page broke, with a `InvalidArgumentException : Method must be a non-empty string. ` error. I'm not sure exactly what caused this issue but I have attached a screenshot of the error page as well as the yaml file which potentially caused the problem.
<img width="1437" alt="Screenshot 2021-02-19 at 16 12 20" src="https://user-images.githubusercontent.com/23292626/108536860-d8270080-72d4-11eb-815d-c84a1d99c525.png">
[TestCase-Request Merchant Payment Reversal – Failure due .txt](https://github.com/gsmainclusivetechlab/interop-test-platform/files/6011691/TestCase-Request.Merchant.Payment.Reversal.Failure.due.txt)
